StateNet/Network/StateNetworkResult.cs

```diff
@@ -12,7 +12,7 @@ private StateNetworkResult(string message, bool success, StateNetwork? network)
         public string Message { get; }
         public bool Success { get; }
         public StateNetwork? Network { get; }
-        public static StateNetworkResult Fail(string message) => new StateNetworkResult(message, false, null);
+        public static StateNetworkResult Fail(StateNetwork? network, string message) => new StateNetworkResult(message, false, network);
```

Comment:
I don't think this is a good idea, you could potentially ignore the success value and access a invalid network.